src/term_rst_post/filetools.py

```diff
@@ -0,0 +1,185 @@
+#
+# Copyright 2021-2021 Vrije Universiteit Brussel
+#
+# This file is part of term_rst_post,
+# originally created by the HPC team of Vrije Universiteit Brussel (https://hpc.vub.be),
+# with support of Vrije Universiteit Brussel (https://www.vub.be),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/sisc-hpc/term_rst_post
+#
+# term_rst_post is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v3.
+#
+# term_rst_post is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this code.  If not, see <http://www.gnu.org/licenses/>.
+#
+##
+"""
+File handling tools for term_rst_post
+
+@author: Alex Domingo (Vrije Universiteit Brussel)
+"""
+
+import argparse
+import logging
+import os
+
+from pathlib import Path
+
+ABLOG_HTML_DIR = '_website'
+
+logger = logging.getLogger()
+
+
+def valid_dirpath(dirpath):
+    """
+    Validate directory path passed through argparse
+    - dirpath: (string) path
+    """
+    if os.path.isdir(dirpath):
+        return dirpath
+    else:
+        errmsg = f"Directory '{dirpath}' does not exist"
+        raise argparse.ArgumentTypeError(errmsg)
+
+
+def fileobj_extension(fileobj):
+    """
+    Return extension (if any) of file object
+    - fileobj: file object
+    """
+    filepath = Path(fileobj.name)
+
+    return filepath.suffix
+
+
+def change_file_extension(filepath, extension, full_path=False):
+    """
+    Return file name changing its extension
+    - filepath: (string) realative or absolute path
+    - extension: (string) new extension of file
+    - full_path: (boolean) return full path (True) or file name (False)
+    """
+    new_filepath = Path(filepath).with_suffix(extension)
+
+    if not full_path:
+        new_filepath = new_filepath.name
+
+    logger.debug(f"File path '{new_filepath}' with extension '{extension}' generated from path '{filepath}'")
+
+    return new_filepath
+
+
+def direct_path(rel_path):
+    """
+    Return path without special maps
```

Comment:
doesn't python have a function for that? abs path or so?

Reply:
this function is actually quite dumb, the goal is just to remove any leading special maps from the path, not to resolve the path. I updated the function renaming it to `inward_path` to make it more clear. You can check the test `test_inward_path` in `test_filetools.py` to see some examples of expected results.